### PR TITLE
Adding hasSameBinaryContentAs

### LINKED ISSUE #1059: Comparing equal binary files fails with java.nio.charset.MalformedInputException
Comparing equal binary files fails with a "java.nio.charset.MalformedInputException: Input length = 1" exception. Method that fails: "FileAssert.hasSameContentAs(java.io.File)".

This is actually caused by a wrong 'fall back' in the 'org.assertj.core.internal.Files' class:
```Java
BinaryDiffResult binaryDiffResult = binaryDiff.diff(actual, readAllBytes(expected.toPath()));
if (binaryDiffResult.hasNoDiff()) {
    // fall back to the RuntimeIOException : not throwing an error is wrong as there was one in the first place.
   throw e;
}
```
In fact this only means the both files are binary qual and it should be:
```Java
BinaryDiffResult binaryDiffResult = binaryDiff.diff(actual, readAllBytes(expected.toPath()));
if (binaryDiffResult.hasNoDiff()) {
   // Both files are binary equal
   return;
}
```
I added a [pull request](https://github.com/joel-costigliola/assertj-core/pull/1060) with the fix and added a test.

### PR DESCRIPTION
**Pull request by ngeor**
#### Check List:
* Fixes #1059 
* Unit tests : YES
* Javadoc with a code example (on API only) : YES

### CODE REVIEW

File: src/test/java/org/assertj/core/api/file/FileAssert_hasSameBinaryContentAs_Test.java [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+    assertThat(actual).hasSameBinaryContentAs(expected);
+  }
+
+  @Test
+  public void should_fail_on_different_files() throws Exception {
+    File actual = createDeleteOnExitTempFileWithContent("assertJ");
+    File expected = createDeleteOnExitTempFileWithContent("assertJ++");
+    assertThatExceptionOfType(AssertionError.class).isThrownBy(() -> {
+      assertThat(actual).hasSameBinaryContentAs(expected);
+    }).withMessageContaining("does not have expected binary content at offset");
```
Comment (joel-costigliola) [inline src/test/java/org/assertj/core/api/file/FileAssert_hasSameBinaryContentAs_Test.java]: prefer `expectAssertionError` as per https://github.com/joel-costigliola/assertj-core/blob/master/CONTRIBUTING.md
add GIVEN WHEN THEN sections

File: src/test/java/org/assertj/core/api/file/FileAssert_hasSameBinaryContentAs_Test.java [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+  }
+
+  private File createDeleteOnExitTempFileWithContent(String content) throws IOException {
```
Comment (joel-costigliola) [inline src/test/java/org/assertj/core/api/file/FileAssert_hasSameBinaryContentAs_Test.java]: rename to `createTempFileWithContent`

**Comment by scordio** [comment]
Closing and reopening to trigger Travis again

**Comment by ngeor** [comment]
> we need to deprecate `hasSameContentAs` for `hasSameTextualContentAs` to fix #1059 otherwise
> just some minor comments, thanks @ngeor!

Thanks, I  processed the minor code changes (I should've read the Contributing guidelines first).
Just to double check, shall I introduce this new method `hasSameTextualContentAs`?

**Comment by joel-costigliola** [comment]
Yes let's add `hasSameTextualContentAs` (a copy of `hasSameContentAs`)

**Comment by ngeor** [comment]
I added the new method and marked the old ones as deprecated. Some extra things:
- I added a `@since` tag to the new methods
- I added a `TempFileUtil` class because the `createTempFile` method was copy pasted around
- Fixed a typo in the docs `Turkisch -> Turkish`

**Comment by joel-costigliola** [comment]
Integrated thanks @ngeor! (I have added the same assertion for path)

**Pull request closed without merging**